### PR TITLE
Extend HydroState to include MHD-vars for HLLD solver

### PR DESCRIPTION
**Pull request by AstroKriel**
I have updated the HLLD solver to make reference to the HydroState struct, rather than the MHDState struct (which I have deleted). I have added the two transverse components of b-field to HydroState, and have also added static_assert's to make sure MHD problems only ever use the HLLD solver.

### CODE REVIEW

**Comment by BenWibking** [comment]
Clang-tidy remains unhappy: https://github.com/quokka-astro/quokka/actions/runs/6451745075/job/17512771179?pr=420#step:4:1100

The macOS build is failing due to an unrelated problem with the Python installation.

**Review by markkrumholz — DISMISSED** [review]

**Comment by BenWibking** [comment]
The clang-tidy check did not pass. Please do not merge until this is fixed.

File: src/HydroState.hpp [hunk 1]
```diff
@@ -14,6 +14,8 @@ template <int Nall, int Nmass> struct HydroState {
 	double cs;				   // adiabatic sound speed
 	double E;				   // total energy density
 	double Eint;				   // internal energy density
+	double by;				   // transverse bfield component
+	double bz;				   // 2nd transverse bfield density
 	std::array<double, Nall> scalar;	   // passive scalars
 	amrex::GpuArray<double, Nmass> massScalar; // mass scalars
```
[the remark this answers is not in the document]
Comment (BenWibking) [reply src/HydroState.hpp:22]: Clang-Tidy is complaining about this. It should be able to be fixed by adding `#include "AMReX.H"` to the list of header includes.

File: src/HydroState.hpp [hunk 2]
```diff
@@ -14,6 +16,8 @@ template <int Nall, int Nmass> struct HydroState {
...
```
[the remark this answers is not in the document]
Comment (BenWibking) [reply src/HydroState.hpp:22]: @AstroKriel Can you add `#include "AMReX_Array.H"`?

[the remark this answers is not in the document]
Comment (BenWibking) [reply src/HydroState.hpp:22]: Ah, I see you already did this. I'll resolve this comment.

File: src/hydro_system.hpp [hunk 3]
```diff
@@ -1026,6 +1028,8 @@ void HydroSystem<problem_t>::ComputeFluxes(amrex::MultiFab &x1Flux_mf, amrex::Mu
 		sL.cs = cs_L;
 		sL.E = E_L;
 		sL.Eint = Eint_L;
+		sL.by = 0.0;
+		sL.bz = 0.0;
```
Comment (BenWibking) [inline src/hydro_system.hpp:1034]: Can you add a comment here that explains that you've just set the magnetic fields to zero for testing purposes?

File: src/hydro_system.hpp [hunk 4]
```diff
@@ -1036,6 +1040,8 @@ void HydroSystem<problem_t>::ComputeFluxes(amrex::MultiFab &x1Flux_mf, amrex::Mu
 		sR.cs = cs_R;
 		sR.E = E_R;
 		sR.Eint = Eint_R;
+		sR.by = 0.0;
+		sR.bz = 0.0;
```
Comment (BenWibking) [inline src/hydro_system.hpp:1047]: Same comment as above -- can you add a comment in the source here that explains that you've just set the magnetic fields to zero for testing purposes?

File: src/hydro_system.hpp [hunk 5]
```diff
@@ -1072,9 +1078,12 @@ void HydroSystem<problem_t>::ComputeFluxes(amrex::MultiFab &x1Flux_mf, amrex::Mu
 		quokka::valarray<double, nvar_> F_canonical{};
 
 		if constexpr (RIEMANN == RiemannSolver::HLLC) {
+			static_assert(!Physics_Traits<problem_t>::is_mhd_enabled, "Cannot use HLLC solver for MHD problems!");
 			F_canonical = quokka::Riemann::HLLC<problem_t, nscalars_, nmscalars_, nvar_>(sL, sR, gamma_, du, dw);
 		} else if constexpr (RIEMANN == RiemannSolver::LLF) {
 			F_canonical = quokka::Riemann::LLF<problem_t, nscalars_, nmscalars_, nvar_>(sL, sR);
+		} else if constexpr (RIEMANN == RiemannSolver::HLLD) {
+			F_canonical = quokka::Riemann::HLLD<problem_t, nscalars_, nmscalars_, nvar_>(sL, sR, gamma_, 0.0);
```
Comment (BenWibking) [inline src/hydro_system.hpp:1091]: Can you add a note that you've hard-coded `bx` to zero for testing purposes?

**Review by BenWibking — CHANGES_REQUESTED** [review]

**Review by BenWibking — APPROVED** [review]

**Pull request merged**